pytorch_lightning/utilities/parsing.py

```diff
@@ -115,11 +128,12 @@ def get_init_args(frame) -> dict:
     self_var, args_var, kwargs_var = parse_class_init_keys(cls)
     filtered_vars = [n for n in (self_var, args_var, kwargs_var) if n]
     exclude_argnames = (*filtered_vars, '__class__', 'frame', 'frame_args')
-
     # only collect variables that appear in the signature
     local_args = {k: local_vars[k] for k in init_parameters.keys()}
     local_args.update(local_args.get(kwargs_var, {}))
     local_args = {k: v for k, v in local_args.items() if k not in exclude_argnames}
+    if OMEGACONF_AVAILABLE:
+        local_args = apply_to_collection(local_args, DictConfig, resolve_dict_config)
```

Comment:
1. if apply_to_collection is recursive, this is exponentially expensive because OmegaConf.to_container is also recursive.
2. can you explain why you need resolve all interpolations eagerly? there are subtle implications to doing it which can be surprising.

Reply:
Because it is currently breaking if I don't do so.

Reply:
`apply_to_collection` is recursive. However, it will stop and apply `resolve_dict_config ` for each time `DictConfig `. In the example, it will apply it directly on cfg as it is the only element in local_args.
If you have a better fix, please submit a PR.

Reply:
What is `subtle implications to doing it which can be surprising.` ? You mean if a user change a value within the configuration. Yes, it will be lost, but we assume inputs to the model are frozen, so we can re-generate it from them.

Reply:
> The interpolation is broken, and need to be resolved.

Why is it broken? OmegaConf containers has a reference to their parent. The interpolation should work:

```python
from omegaconf import *

cfg = OmegaConf.create({"foo": {"bar": "${zoo}"}, "zoo": 20})

assert cfg.foo.bar == 20
foo = cfg.foo
assert foo.bar == 20
```
Something was probably done along the way to break it. 
I think it's better to identify what is the root cause before fixing this.

> If you have a better fix, please submit a PR.

You seem a bit defensive, If you don't need my advice you are welcome to ignore it.

Reply:
Hey @omry

ahaha I apologise, I didn't mean to sound defensive. Your advice are always welcome !!!
I meant `The interpolation is broken, and we need to resolve it` on the  Lightning side. We are doing something wrong ! 👍 

And yes, you are entirely right.  @SeanNaren spotted the root cause of the problem.
We were off in our loop. Surprising our tests didn't catch this ...

Just pushed @SeanNaren fix and test pass.

Reply:
Accepted, I think it was just a small misunderstanding.

When I asked "Why is it broken?" I did not think you meant that it's broken in OmegaConf, but rather that it's not working correctly. My question was more along the lines of "Why is it not working correctly? there has to be something going on here".

Anyway, water under the bridge. Great to see you improving Hydra support in PL!